osu.Game/OsuGame.cs

```diff
@@ -178,6 +180,30 @@ private void load(FrameworkConfigManager frameworkConfig)
         /// <param name="setId">The set to display.</param>
         public void ShowBeatmapSet(int setId) => beatmapSetOverlay.FetchAndShowBeatmapSet(setId);
 
+        /// <summary>
+        /// Present a beatmap at song select.
+        /// </summary>
+        /// <param name="beatmap">The beatmap to select.</param>
+        public void PresentBeatmap(BeatmapSetInfo beatmap)
+        {
+            CloseAllOverlays();
+
+            Beatmap.Value = BeatmapManager.GetWorkingBeatmap(beatmap.Beatmaps.First());
```

Comment:
Crashes if the notification is clicked during gameplay, and in the editor. Both probably should be disallowed.